File: gatorMoisture.ts
/**
* Mary West @ SparkFun Electronics
* July 3, 2017
* https://github.com/sparkfun/pxt-gator-temp

* Development environment specifics:
* Written in Microsoft Makecode
* Tested with a SparkFun gatortemp sensor and micro:bit
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/


/**
 * Functions to operate the gatortemp sensor
 */

 enum gatorMoistureType{
  moisture=1,
  adcVal=2,
  Test = 3,
}

//% color=#8fce00 icon="\f7b4"
namespace gatorMoisture {

  // Functions for reading moisture from the gatormoisture in moisture or straight adv value

  /**
  * Reads the number
  */
  //% weight=30 blockId="gatorMoisture_moisture" block="Get moisture on pin %pin | in %gatorMoistureType"
  export function moisture(pin: AnalogPin, type: gatorMoistureType): number{
      let ADCVal = pins.analogReadPin(pin)
      switch(type){
          case gatorMoistureType.moisture: return getMoisture(ADCVal)
          case gatorMoistureType.adcVal: return ADCVal
          case gatorMoistureType.Test: return ADCVal
          default: return -11111111
      }
  }

  /**
   * Function used for simulator, actual implementation is in gatormoisture.cpp
   */
  //% shim=gatorMoisture::getMoisture
  function getMoisture(ADCVal: number) {
      // Fake function for simulator
      return 0
  }
}

File: gatorMoisture.cpp

/**
* Mary West @ SparkFun Electronics
* Juuly 3, 2017
* https://github.com/sparkfun/pxt-gator-temp
*
* Development environment specifics:
* Written in Microsoft PXT
* Tested with a SparkFun MCP9700 sensor and micro:bit
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/

#include "pxt.h"
#include <cstdint>
#include <math.h>

using namespace pxt;

namespace gatorMoisture {
    /*
    * Calculates the moisture based on the ADC value passed in. 1 step in adcVal is equal to .488 uA or .976 lux at 5V
    */
    //%
    uint16_t getMoisture(int16_t ADCVal) {
        return ADCVal * 1023.0;
    }

}

